app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -79,7 +92,67 @@ def package_version
         JSON.parse(package_json.read)['version']
       end
 
+      def publish_config
+        JSON.parse(package_json.read)['publishConfig']
+      end
+
+      def publish_config_access
+        publish_config['access'] || PRIVATE
+      end
+
+      def package_name
+        JSON.parse(package_json.read)['name']
+      end
+
+      def scoped_package?
+        package_name.start_with?('@shopify')
+      end
+
+      def publish?
+        return false if publish_config.blank?
+        return false unless valid_publish_config_access?
+        return false unless package_scoped_when_private?
+      end
+
+      def valid_publish_config_access?
+        return false unless VALID_ACCESS.include?(publish_config_access)
+        true
+      end
+
+      def package_scoped_when_private?
+        private_package = publish_config_access == PRIVATE
+        return false if private_package && !scoped_package?
```

Comment:
Isn't this the same as `private_package && scoped_package?`?

`package_scoped_when_private?` reads oddly.  `private_scoped_package?` makes more sense to me.

Reply:
> package_scoped_when_private? reads oddly. private_scoped_package? makes more sense to me.

`package_scoped_when_private?` ensures that the package is scoped when it's private. `private_scoped_package?` reads as if the package must be private and scoped. 

> Isn't this the same as private_package && scoped_package??

`private_package && scoped_package?` will return false if the package is public and fail `publish?`. I've added an additional test to cover this case. 

I'll comment back once it's pushed.

Reply:
Actually, I can do something like this, and it reads better:

```
return true if publish_config_access == PUBLIC
publish_config_access == PRIVATE && scoped_package?
```